src/file/match.cr

```diff
@@ -10,20 +10,18 @@ class File < IO::FileDescriptor
   #
   # The pattern syntax is similar to shell filename globbing. It may contain the following metacharacters:
   #
-  # * `*` matches an unlimited number of arbitrary characters, excluding any directory separators.
-  #   * `"*"` matches all regular files.
-  #   * `"c*"` matches all files beginning with `c`.
-  #   * `"*c"` matches all files ending with `c`.
-  #   * `"*c*"` matches all files that have `c` in them (including at the beginning or end).
-  # * `**` matches directories recursively if followed by `/`.
-  #   If this path segment contains any other characters, it is the same as the usual `*`.
-  # * `?` matches one arbitrary character, excluding any directory separators.
+  # * `*`: Wildcard matches zero or more characters, except for directory separators.
+  # * `**`: Globstar matches zero or more characters, including directory separators.
+  #   It must match a complete path segment, i.e. it must be wrapped in `/`
+  #   except for the beginning and end of the pattern.
+  # * `?`: Matches a single Unicode character, except for directory separators.
   # * character sets:
```

Comment:
Super nitpick but suggest: `* Character sets:` (on a first read I thought this line was accidentally skipped in the changeset)

Reply:
Nevermind my previous comment, I was reviewing commit by commit and I saw you fixed it in a subsequent commit